### ISSUE TITLE
[suggestion][website] Add support of private plugins

### ISSUE THREAD

**Issue by AmazingBeu** [openplanet-nl/openplanet-issues#600]
Some of us could need to have private plugins, reserved for specific users for many reasons (privacy, secret training, security, ...)

Currently, the website allows us to create a plugin and never release it. But there some thing that could be improved:
- Allow to publish plugin in private (to fix the diff mode for the reviews according XertroV)
- Hide the plugin name on https://openplanet.dev/plugins/signs
- Add a way to allow specific users to see, download and update the plugin

I thought about 2 ways to allow users:
- Add an "Allow list" with the login of the allowed users. That will require to add authentication on the plugin manager plugin
- Add a secret key per plugin, renewable in case of leak, and the plugin manager will have to add the keys in all its requests, and you *just* have to fetch plugin by key server side

**Comment by codecat** [comment]
I don't think we should be hiding plugin names in the signing list, because then we would not be transparent anymore in what we are signing.

**Comment by AmazingBeu** [comment]
Not really a problem imo, especially if it's noted when you publish a private plugin
